gdal/doc/source/api/index.rst

```diff
@@ -85,6 +85,7 @@ API
 
        python_api_ref
        python_gotchas
+       python_samples
```

Comment:
Note unrelated to that PR. The doc structure is a bit messy regarding python. We have that section, and if we go a bit further in the page, we also have a dedicated page for python. I'm not sure how, but I feel like this should be unified as the current setup is confusing.

Reply:
like that https://github.com/OSGeo/gdal/pull/3687 ?